### PR TITLE
Better default ports

### PR DESCRIPTION
**Pull request by mixmix**
The purpose of this PR is to : 
- :heavy_check_mark:  ensure net + ws ports are set to default if not specified in `connections.incoming`
  - note this only fixes the "primary" connection in each case, does not fix multiple net connections missing `port`
- :heavy_check_mark: **ensure `host` and `port` are _not_ set** in the event the user explicitly did not declare an incoming net or ws connection

TODO:
- _ test in ssb-server

### CODE REVIEW

**Comment by mixmix** [comment]
@dominictarr this is to address  6fd058f70183bee25afb949806700c7398dbcbd4 : 
> net and ws should always have a default port, otherwise breaks ssb-server/plugins/local, @mixmix

`ssb-config` should allow you to _not set port when you're not wanting one set. If `ssb-server` crashes without a port, then we need to either: 
- ensure there is a transport from which we're getting a valid `port`
- teach `ssb-server` to not explode

_example edge case : I start an `ssb-server` with only a socket connection. `ssb-server` should need no port... and if it does, then I'm suspicious about what it's doing with it_

File: tests/connections.js [hunk 1]
```diff
@@ -102,3 +102,54 @@ test('CONFLICTING custom host:port connections.incoming settings', t => {
...
+    connections: {
+      incoming: {
+        net: [{ host: 'pub.mixmix.io', scope: 'public' }]
+      }
+    }
+  })
+
+  var { port } = config.connections.incoming.net[0]
+  t.equal(port, 8008, 'net: sets default port in connections')
+  t.equal(config.port, 8008, 'net: [LEGCACY] default config.port is set')
```
Comment (hamnox) [inline tests/connections.js]: L136 & L152: s/LEGCACY/LEGACY

**Comment by dominictarr** [comment]
:+1:
